roachpb/errors.go

```diff
@@ -275,6 +275,9 @@ func (e *TransactionAbortedError) Transaction() *Transaction {
 func NewTransactionPushError(txn, pusheeTxn Transaction) *TransactionPushError {
 	err := &TransactionPushError{PusheeTxn: *pusheeTxn.Clone()}
 	if len(txn.ID) != 0 {
+		// Set only when the Txn ID is set since a pusher Txn
+		// can be a fake one that contains priority without ID
+		// (see Store.resolveWriteIntentError).
```

Comment:
more concise and without leaking implementation details:

// When the pusher is non-transactional, txn will be empty but for the priority. In that case, ignore it here.